src/Kdyby/Doctrine/Connection.php

```diff
@@ -141,94 +143,21 @@ public function insert($tableExpression, array $data, array $types = [])
 
 
 	/**
-	 * @param string $query
-	 * @param array $params
-	 * @param array $types
-	 * @param \Doctrine\DBAL\Cache\QueryCacheProfile $qcp
-	 * @return \Doctrine\DBAL\Driver\Statement
-	 * @throws DBALException
-	 */
-	public function executeQuery($query, array $params = [], $types = [], Doctrine\DBAL\Cache\QueryCacheProfile $qcp = NULL)
-	{
-		try {
-			return parent::executeQuery($query, $params, $types, $qcp);
-
-		} catch (\Exception $e) {
-			throw $this->resolveException($e, $query, $params);
-		}
-	}
-
-
-
-	/**
-	 * @param string $query
-	 * @param array $params
-	 * @param array $types
-	 * @return int
-	 * @throws DBALException
-	 */
-	public function executeUpdate($query, array $params = [], array $types = [])
-	{
-		try {
-			return parent::executeUpdate($query, $params, $types);
-
-		} catch (\Exception $e) {
-			throw $this->resolveException($e, $query, $params);
-		}
-	}
-
-
-
-	/**
-	 * @param string $statement
-	 * @return int
-	 * @throws DBALException
-	 */
-	public function exec($statement)
-	{
-		try {
-			return parent::exec($statement);
-
-		} catch (\Exception $e) {
-			throw $this->resolveException($e, $statement);
-		}
-	}
-
-
-
-	/**
-	 * @return \Doctrine\DBAL\Driver\Statement|mixed
-	 * @throws DBALException
-	 */
-	public function query()
-	{
-		$args = func_get_args();
-		try {
-			return call_user_func_array('parent::query', $args);
-
-		} catch (\Exception $e) {
-			throw $this->resolveException($e, func_get_arg(0));
-		}
-	}
-
-
-
-	/**
 	 * Prepares an SQL statement.
 	 *
 	 * @param string $statement The SQL statement to prepare.
 	 * @throws DBALException
-	 * @return PDOStatement The prepared statement.
+	 * @return Statement The prepared statement.
 	 */
 	public function prepare($statement)
 	{
 		$this->connect();
```

Comment:
It is indeed unneccesary. It was here only because older version of doctrine had it.